Editor/AGS.Editor/Panes/FreePanControl.cs

```diff
@@ -0,0 +1,98 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Drawing;
+using System.Text;
+using System.Windows.Forms;
+
+namespace AGS.Editor
+{
+    public class FreePanControl : BufferedPanel
+    {
+        private const int WS_EX_COMPOSITED = 0x02000000;
+        private int _lastX, _lastY; //used to prevent useless mouse move events
+        private Point _panGrabPoint = Point.Empty;
+        private Cursor _panCursor = Cursors.Hand; // We need a better hand cursor, using the "link" hand for now
+        public event EventHandler PanGrabbed;
+        public event EventHandler PanReleased;
+
+        // Enabling Composite makes scrolling smoother
+        protected override CreateParams CreateParams
+        {
+            get
+            {
+                CreateParams cp = base.CreateParams;
+                cp.ExStyle |= WS_EX_COMPOSITED;
+                return cp;
+            }
+        }
+
+        public FreePanControl() : base()
+        {
+
+        }
+
+        protected virtual void OnPanGrab(EventArgs e)
+        {
+            PanGrabbed?.Invoke(this, EventArgs.Empty);
+        }
+
+        protected virtual void OnPanRelease(EventArgs e)
+        {
+            PanReleased?.Invoke(this, EventArgs.Empty);
+        }
+
+        protected override void OnMouseDown(MouseEventArgs e)
+        {
+            // defines the shortcut for click pan Control+Shift+Left Click Hold or Middle Click Hold without any modifier.
+            if ( (e.Button == MouseButtons.Left   && Control.ModifierKeys == (Keys.Control|Keys.Shift)) ||
```

Comment:
```suggestion
            if ( (e.Button == MouseButtons.Left   && ModifierKeys == (Keys.Control|Keys.Shift)) ||
```

Reply:
updated to apply this simplification. Bonus Control doesn't confuse with Keys.Control, so a good improvement!